File: spring-test/src/main/java/ru/mydesignstudio/spring/test/web/service/impl/PersonServiceImpl.java
package ru.mydesignstudio.spring.test.web.service.impl;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;
import ru.mydesignstudio.spring.test.web.model.Person;
import ru.mydesignstudio.spring.test.web.service.PersonService;

import java.util.Collection;

@Service
public class PersonServiceImpl implements PersonService {
    @Autowired
    private PersonDao personDao;

    @Override
    public Person findOne(int id) {
        return personDao.findOne(id);
    }

    @Override
    public Person save(Person person) {
        return personDao.save(person);
    }

    @Override
    public Collection<Person> findAll() {
        return personDao.findAll();
    }
}

File: kotlin-samples/src/main/kotlin/ru/mydesignstudio/kotlin/samples/person/PersonServiceImpl.kt
package ru.mydesignstudio.kotlin.samples.person

class PersonServiceImpl(private val personDao: PersonDao): PersonService {
    override fun findAll(): Collection<Person> = personDao.findAll()

    override fun findPerson(id: Int): Person = personDao.findPerson(id)

    override fun savePerson(person: Person): Person = personDao.savePerson(person)

    override fun delete(person: Person) = personDao.delete(person)
}
